src/__tests__/lists.ts

```diff
@@ -0,0 +1,257 @@
+  `
+  const context = mockContext(3)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(obj).toMatchSnapshot()
+    expect(parseError(context.errors)).toBe('')
+    expect(obj.status).toBe('finished')
+    expect((obj as Finished).value).toBe(true)
+  })
+})
+
+test('member', () => {
+  const code = `
+    equal(member("string", list(1, 2, 3, "string", 123, 456, null, undefined)), list("string", 123, 456, null, undefined));
+  `
+  const context = mockContext(3)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(obj).toMatchSnapshot()
+    expect(parseError(context.errors)).toBe('')
+    expect(obj.status).toBe('finished')
+    expect((obj as Finished).value).toBe(true)
+  })
+})
+
+test('remove', () => {
+  const code = `
+    remove(1, list(1));
+  `
+  const context = mockContext(3)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(obj).toMatchSnapshot()
+    expect(parseError(context.errors)).toBe('')
+    expect(obj.status).toBe('finished')
+    expect((obj as Finished).value).toBe(null)
+  })
+})
+
+test('remove_all', () => {
+  const code = `
+    equal(remove_all(1, list(1, 2, 3, 4, 1, 1, "1", 5, 1, 1, 6)), list(2, 3, 4, "1", 5, 6));
+  `
+  const context = mockContext(3)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(obj).toMatchSnapshot()
+    expect(parseError(context.errors)).toBe('')
+    expect(obj.status).toBe('finished')
+    expect((obj as Finished).value).toBe(true)
+  })
+})
+
+test('enum_list', () => {
+  const code = `
+    equal(enum_list(1, 5), list(1, 2, 3, 4, 5));
+  `
+  const context = mockContext(3)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(obj).toMatchSnapshot()
+    expect(parseError(context.errors)).toBe('')
+    expect(obj.status).toBe('finished')
+    expect((obj as Finished).value).toBe(true)
+  })
+})
+
+test('list_ref', () => {
+  const code = `
+    list_ref(list(1, 2, 3, "4", 4), 4);
+  `
+  const context = mockContext(3)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(obj).toMatchSnapshot()
+    expect(parseError(context.errors)).toBe('')
+    expect(obj.status).toBe('finished')
+    expect((obj as Finished).value).toBe(4)
+  })
+})
+
+test('accumulate', () => {
+  const code = `
+    accumulate((curr, acc) => curr + acc, 0, list(2, 3, 4, 1));
+  `
+  const context = mockContext(3)
+  const promise = runInContext(code, context, { scheduler: 'preemptive' })
+  return promise.then(obj => {
+    expect(obj).toMatchSnapshot()
+    expect(parseError(context.errors)).toBe('')
+    expect(obj.status).toBe('finished')
+    expect((obj as Finished).value).toBe(10)
+  })
+})
+
```

Comment:
You should test `equal` as well if not `equal` returning `true` for all input would actually pass all the test cases :o